matrix/matrix_operation.py

```diff
@@ -2,38 +2,46 @@
 function based version of matrix operations, which are just 2D arrays
 """
 
+from typing import List, Tuple
 
-def add(matrix_a, matrix_b):
+
+def add(matrix_a: List[list], matrix_b: List[list]) -> List[list]:
```

Comment:
`List[list]` or `List[List[int]]`?  The first would allow us to do floats instead of just ints.  If the former then please add doctests that use floats.

Reply:
added.